third_party/move/move-vm/runtime/src/module_traversal.rs

```diff
@@ -117,6 +118,37 @@ impl<'a> TraversalContext<'a> {
     }
 }
 
+impl<'a> ModuleTraversalContext for TraversalContext<'a> {
+    fn visit_if_not_special_module_id(&mut self, module_id: &ModuleId) -> PartialVMResult<bool> {
+        let addr = module_id.address();
+        if addr.is_special() {
+            return Ok(false);
+        }
+
+        let name = module_id.name();
+        Ok(if self.visited.contains_key(&(addr, name)) {
+            false
+        } else {
+            let module_id = self.referenced_module_ids.alloc(module_id.clone());
+            self.visited
```

Comment:
why is visited a BTreeMap with () value type and not a BTreeSet?

Reply:
I had the same question, it was like that before when first introduced, @vgao1996 should have more context.